app/src/main/java/au/com/bluedot/minimalintegration/MainApplication.java

```diff
@@ -62,17 +44,27 @@ public void initPointSDK() {
 
         boolean locationPermissionGranted =
                 ActivityCompat.checkSelfPermission(getApplicationContext(), Manifest.permission.ACCESS_FINE_LOCATION) == PackageManager.PERMISSION_GRANTED;
-        boolean backgroundPermissionGranted = (Build.VERSION.SDK_INT < Build.VERSION_CODES.Q)
-                || ActivityCompat.checkSelfPermission(getApplicationContext(), Manifest.permission.ACCESS_BACKGROUND_LOCATION) == PackageManager.PERMISSION_GRANTED;
 
-        if (locationPermissionGranted && backgroundPermissionGranted) {
+        if (locationPermissionGranted) {
             mServiceManager = ServiceManager.getInstance(this);
 
-            if (!mServiceManager.isBlueDotPointServiceRunning()) {
-                // Setting Notification for foreground service, required for Android Oreo and above.
-                // Setting targetAllAPIs to TRUE will display foreground notification for Android versions lower than Oreo
-                mServiceManager.setForegroundServiceNotification(createNotification(), false);
-                mServiceManager.sendAuthenticationRequest(apiKey, this, restartMode);
+            if (!mServiceManager.isBluedotServiceInitialized()) {
+
+                InitializationResultListener resultListener = bdError -> {
+                    String text = "Initialization Result ";
+                    if(bdError != null)
+                        text = text + bdError.getReason();
+                    else {
+                        text = text + "Success ";
+                        Intent intent = new Intent(getApplicationContext(), MainActivity.class);
+                        intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
+                        startActivity(intent);
+                    }
+                    Toast.makeText(getApplicationContext(),text,Toast.LENGTH_LONG).show();
+                };
+                //ProjectID for the App 
+                String projectId = "<YOUR-PROJECT-ID>";
```

Comment:
Let's make this a constant right at the top of the file so it's harder to miss

Reply:
moved projectId and destinationId to tthe op of the class